src/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -222,14 +254,7 @@ private void CreateFromToken(IntPtr userToken)
             if (Marshal.GetLastWin32Error() == Interop.Errors.ERROR_INVALID_HANDLE)
                 throw new ArgumentException(SR.Argument_InvalidImpersonationToken);
 
-            if (!Interop.Kernel32.DuplicateHandle(Interop.Kernel32.GetCurrentProcess(),
-                                             userToken,
-                                             Interop.Kernel32.GetCurrentProcess(),
-                                             ref _safeTokenHandle,
-                                             0,
-                                             true,
-                                             Interop.DuplicateHandleOptions.DUPLICATE_SAME_ACCESS))
-                throw new SecurityException(new Win32Exception().Message);
+            _safeTokenHandle = DuplicateAccessToken(userToken);
```

Comment:
Does the new handle get disposed of correctly eventually?

Reply:
The handle will be closed once the GC determines there are no more references and the SafeHandle is finalized

Reply:
So we're depending on finalization.  Is that how things work in netfx, too?

Reply:
Yea, there's no other clear way of identifying the proper lifetime for the handle from the API side